M/international/locations/states forms

RaimondasSapola opened the pull request:
<!--
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
 -->

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop / 1.7.6.x / 1.6.1.x
| Description?  | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc.
| Type?         | bug fix / improvement / new feature / refacto / critical
| Category?     | FO / BO / CO / IN / WS / TE
| BC breaks?    | Does it break backward compatibility? yes/no
| Deprecations? | Does it deprecate an existing feature? yes/no
| Fixed ticket? | (optional) If this PR fixes an [Issue](https://github.com/PrestaShop/PrestaShop/issues), please write "Fixes #[issue number]" here.
| How to test?  | Please indicate how to best verify that this PR is correct.

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

RaimondasSapola committed:
WIP forms

RaimondasSapola committed:
Custom validator, updated controller with new error messages

RaimondasSapola committed:
INVALID_ACTIVE -> INVALID_STATE

RaimondasSapola committed:
creation state handling and legacy index route

RaimondasSapola committed:
missing php docs and removed unneeded js import

RaimondasSapola committed:
Updated translations

RaimondasSapola committed:
Fixed tests issues for country choices provider

sarjon commented on src/Adapter/State/AbstractStateHandler.php:
```diff
@@ -0,0 +1,168 @@
+        try {
+            $state = new State($stateIdValue);
+        } catch (PrestaShopException $e) {
+            throw new StateNotFoundException(
```
If constructor of legacy object fails, I don't think it's a `not found` exception, since this could indicate that database is down. Wdyt?

sarjon commented on src/Adapter/State/AbstractStateHandler.php:
```diff
@@ -0,0 +1,168 @@
+     *
+     * @throws CountryNotFoundException
+     */
+    protected function assertCountryWithIdExists(int $id)
```
Why not passing `CountryId` VO?

sarjon commented on src/Adapter/State/AbstractStateHandler.php:
```diff
@@ -0,0 +1,168 @@
+        try {
+            $country = new Country($id);
+        } catch (PrestaShopException $e) {
+            throw new CountryNotFoundException(
```
Same here, it is really `not found` exception?

sarjon commented on src/Adapter/State/AbstractStateHandler.php:
```diff
@@ -0,0 +1,168 @@
+     *
+     * @throws ZoneNotFoundException
+     */
+    protected function assertZoneWithIdExists(int $id)
```
Same comments for this method argument and `not found` exception.

sarjon commented on src/Adapter/State/AbstractStateHandler.php:
```diff
@@ -0,0 +1,168 @@
+     */
+    protected function assertFieldContainsCleanHtml(string $field, int $code)
+    {
+        $errors = $this->validator->validate($field, new CleanHtml());
```
Are you validating a `field` or a `value`? :grin:

sarjon commented on src/Adapter/State/AbstractStateHandler.php:
```diff
@@ -0,0 +1,168 @@
+                $field
+            ),
+                $code
+            );
```
Weird code formatting, don't forget to run `cs-fixer`. :wink:

sarjon commented on src/Adapter/State/AbstractStateHandler.php:
```diff
@@ -0,0 +1,168 @@
+    }
+
+    /**
+     * @param bool $active
```
Why not use actual typehint?

sarjon commented on src/Adapter/State/CommandHandler/AddStateHandler.php:
```diff
@@ -0,0 +1,100 @@
+    public function __construct(ValidatorInterface $validator)
+    {
+        parent::__construct($validator);
+    }
```
Do you need to redeclare construtor here?

sarjon commented on src/Adapter/State/CommandHandler/AddStateHandler.php:
```diff
@@ -0,0 +1,100 @@
+            $state = new State();
+
+            $this->assertFieldContainsCleanHtml($command->getName(), StateConstraintException::INVALID_NAME);
+            $this->assertFieldContainsCleanHtml($command->getIsoCode(), StateConstraintException::INVALID_ISO_CODE);
```
I think you could avoid having this assertion and `ValidatorInterface` dependency. You could use legacy `$objectModel->validateField()` to do the same. :)

sarjon commented on src/Adapter/State/CommandHandler/EditStateHandler.php:
```diff
@@ -0,0 +1,120 @@
+/**
+ * Handles state editing
+ */
+class EditStateHandler extends AbstractStateHandler implements EditStateHandlerInterface
```
Should be `final.

sarjon commented on src/Adapter/State/CommandHandler/EditStateHandler.php:
```diff
@@ -0,0 +1,120 @@
+    private function updateStateFromCommandData(State $state, EditStateCommand $command)
+    {
+        try {
+            if ($command->getZoneId()) {
```
Maybe you could check against `null`? `if (null !== $command->getZoneId()) {}`, I think it's more clear of what you are doing.

sarjon commented on src/Adapter/State/CommandHandler/EditStateHandler.php:
```diff
@@ -0,0 +1,120 @@
+            }
+
+            if ($command->getIsoCode()) {
+                $this->assertFieldContainsCleanHtml($command->getIsoCode(), StateConstraintException::INVALID_ISO_CODE);
```
Same here, you don't really need an assertion, you could use method of legacy object. :slightly_smiling_face:

sarjon commented on src/Adapter/State/QueryHandler/IsUniqueStateIsoCodeHandler.php:
```diff
@@ -0,0 +1,57 @@
+/**
+ * Handles query for determining if state with given iso code exists
+ */
+final class IsUniqueStateIsoCodeHandler implements IsUniqueStateIsoCodeHandlerInterface
```
How about `GetStateWithIsoCodeExistsHandler`? :thinking:

sarjon commented on src/Adapter/State/QueryHandler/IsUniqueStateIsoCodeHandler.php:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(isUniqueStateIsoCode $query): IsFoundStateByIsoCode
```
`IsFoundStateByIsoCode` feels a bit overengineered. :sweat_smile: How about returning just `bool`?

sarjon commented on src/Adapter/State/QueryHandler/IsUniqueStateIsoCodeHandler.php:
```diff
@@ -0,0 +1,57 @@
+            return new IsFoundStateByIsoCode(false);
+        }
+
+        if ($query->getStateId() !== null && $state != $query->getStateId()) {
```
How about using strict comparison?

RaimondasSapola replied:
Strict comparison in this place will always fail because State::getIdByIso($query->getIsoCode()) returns statet id as string and $query->getStateId() returns int. And casting string to int just because strict comparison could be used seemed useless

sarjon commented on src/Adapter/Zone/ZoneDataProvider.php:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * {@inheritdoc}
+     */
+    public function getZones($active = false, $activeFirst = false): array
```
How's that different than legacy? You interface should not be influenced by legacy code, I think this needs to be refactored. :)

sarjon commented on src/Core/Domain/State/Command/AddStateCommand.php:
```diff
@@ -0,0 +1,119 @@
+        int $zoneId,
+        string $name,
+        string $isoCode,
+        ?bool $active
```
Let's avoid nullable types and set `$active` to `false` by default. Wdyt?

sarjon commented on src/Core/Domain/State/Command/AddStateCommand.php:
```diff
@@ -0,0 +1,119 @@
+        ?bool $active
+    ) {
+        $this->countryId = $countryId;
+        $this->zoneId = $zoneId;
```
Both zone id and country id should be made as VOs.

sarjon commented on src/Core/Domain/State/Command/EditStateCommand.php:
```diff
@@ -0,0 +1,181 @@
+    /**
+     * @param StateId $stateId
+     */
+    public function __construct(StateId $stateId)
```
Command should accept scalar value and transform it to VO inside constructor instead.

sarjon commented on src/Core/Domain/State/CommandHandler/AddStateHandlerInterface.php:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * @param AddStateCommand $command
+     */
+    public function handle(AddStateCommand $command);
```
It should return `StateId`.

sarjon commented on src/Core/Domain/State/Exception/StateConstraintException.php:
```diff
@@ -0,0 +1,43 @@
+
+    const INVALID_ISO_CODE = 3;
+
+    const INVALID_FIELDS = 4;
```
I don't think `fields` is right. :/

sarjon commented on src/Core/Domain/State/Query/IsUniqueStateIsoCode.php:
```diff
@@ -0,0 +1,79 @@
+     *
+     * @return isUniqueStateIsoCode
+     */
+    public function setStateId(int $stateId): isUniqueStateIsoCode
```
Why would you set id here? :thinking: 

p.s. `isUniqueStateIsoCode` should start with capital letter.

RaimondasSapola replied:
:(

RaimondasSapola replied:
Set id is for edit state case. In this case state id should be compared with value provided by SearchStateByIsoCode method to make sure validation wouldn't be triggered if id value found is of the edited entity.

sarjon commented on src/Core/Domain/State/QueryResult/EditableState.php:
```diff
@@ -0,0 +1,137 @@
+    public function __construct(
+        StateId $stateId,
+        int $countryId,
+        int $zoneId,
```
You could pass VOs for IDs as well.

sarjon commented on src/Core/Domain/State/QueryResult/IsFoundStateByIsoCode.php:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Transfers boolean value to determine if state by iso code was found
+ */
+class IsFoundStateByIsoCode
```
As discussed above, this class seems redundant.

sarjon commented on src/Core/Domain/State/ValueObject/StateId.php:
```diff
@@ -0,0 +1,74 @@
+     */
+    private function assertPositiveInt(int $value)
+    {
+        if (0 >= $value) {
```
Is `0` considered a valid id?

sarjon commented on src/Core/Domain/Zone/DataProvider/ZoneDataProviderInterface.php:
```diff
@@ -0,0 +1,43 @@
+/**
+ * Defines contract for zone data provider
+ */
+interface ZoneDataProviderInterface
```
It should be moved out of Domain and possibly refactored as discussed in previous comments.

sarjon commented on src/Core/Form/ChoiceProvider/CountryByIdChoiceProvider.php:
```diff
@@ -27,12 +27,12 @@
  * Class CountryByIdChoiceProvider provides country choices with ID values.
  */
-final class CountryByIdChoiceProvider implements FormChoiceProviderInterface
+final class CountryByIdChoiceProvider implements ConfigurableFormChoiceProviderInterface
```
You cannot switch interface in the class, it's a BC break. What you need to do is implement new service instead.

sarjon commented on src/Core/Form/ChoiceProvider/ZoneByIdChoiceProvider.php:
```diff
@@ -0,0 +1,74 @@
+     */
+    public function __construct(ZoneDataProviderInterface $zoneDataProvider)
+    {
+        $this->zoneDataProvider = $zoneDataProvider;
```
Instead to avoid creating this data provider, I suggest you to implement `ZoneByIdChoiceProvider` as adapter, so you can use legacy class in it.

sarjon commented on src/Core/Form/IdentifiableObject/DataHandler/StateFormDataHandler.php:
```diff
@@ -0,0 +1,87 @@
+     */
+    public function create(array $data)
+    {
+        $this->commandBus->handle(new AddStateCommand(
```
Don't forget to return new id.

sarjon commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -58,10 +69,174 @@ public function getStatesAction(Request $request)
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/add.html.twig', [
+            'stateForm' => $attachmentForm->createView(),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
```
Check if `Help` button is working in form page.

sarjon commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -58,10 +69,174 @@ public function getStatesAction(Request $request)
+        );
+
+        $attachmentForm = $attachmentFormBuilder->getForm();
+        $attachmentForm->handleRequest($request);
```
Check naming for all variables, this is not attachment.

sarjon commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -58,10 +69,174 @@ public function getStatesAction(Request $request)
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_states_index"
+     * )
+     * @DemoRestricted(redirectRoute="admin_states_index")
```
Was it demo restricted before?

sarjon commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -58,10 +69,174 @@ public function getStatesAction(Request $request)
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/edit.html.twig', [
+            'stateForm' => null !== $stateForm ? $stateForm->createView() : null,
```
Can form be null in edit page?

sarjon commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -58,10 +69,174 @@ public function getStatesAction(Request $request)
+    {
+        try {
+            /** @var EditableState $stateInformation */
+            $stateInformation = $this->getQueryBus()->handle(new GetStateForEditing((int) $stateId));
```
I'd prefer to call it `$editableState`.

sarjon commented on src/PrestaShopBundle/Form/Admin/Improve/International/Locations/State/StateType.php:
```diff
@@ -0,0 +1,88 @@
+                'required' => true,
+                'constraints' => [
+                    new NotBlank(),
+                ],
```
Don't you need to check any other constraints? Like clean html?

sarjon commented on src/PrestaShopBundle/Form/Admin/Improve/International/Locations/State/StateType.php:
```diff
@@ -0,0 +1,88 @@
+            ])
+            ->add('id_country', CountryChoiceType::class, [
+                'required' => true,
+                'containsStates' => true,
```
Array keys should be `snake_case`.

sarjon commented on src/PrestaShopBundle/Form/Admin/Improve/International/Locations/State/StateType.php:
```diff
@@ -0,0 +1,88 @@
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefault('constraints', new UniqueStateIsoCode());
```
Shouldn't `constraints ` be an array?

RaimondasSapola replied:
It doesn't have to be

sarjon commented on src/PrestaShopBundle/Form/Admin/Type/CountryChoiceType.php:
```diff
@@ -37,14 +38,14 @@
+     * @param ConfigurableFormChoiceProviderInterface $countriesChoiceProvider
      */
-    public function __construct(FormChoiceProviderInterface $countriesChoiceProvider)
+    public function __construct(ConfigurableFormChoiceProviderInterface $countriesChoiceProvider)
```
This is a BC break. :(

sarjon commented on src/PrestaShopBundle/Form/Admin/Type/CountryChoiceType.php:
```diff
@@ -54,13 +55,25 @@ public function __construct(FormChoiceProviderInterface $countriesChoiceProvider
+                    $this->countriesChoiceProvider->getChoices([
+                        'active' => $options['active'],
+                        'containsStates' => $options['containsStates'],
+                        'listStates' => $options['listStates'],
```
`snake_case`.

sarjon commented on src/PrestaShopBundle/Form/Admin/Type/CountryChoiceType.php:
```diff
@@ -54,13 +55,25 @@ public function __construct(FormChoiceProviderInterface $countriesChoiceProvider
-            'choices' => $choices,
+            'active' => false,
+            'containsStates' => false,
+            'listStates' => false,
```
`snake_case`

sarjon commented on src/PrestaShopBundle/Form/Admin/Type/ZoneChoiceType.php:
```diff
@@ -0,0 +1,85 @@
+                    ['--' => ''],
+                    $this->zonesChoiceProvider->getChoices([
+                        'active' => $options['active'],
+                        'activeFirst' => $options['activeFirst'],
```
`snake_case`.

sarjon commented on src/PrestaShopBundle/Form/Admin/Type/ZoneChoiceType.php:
```diff
@@ -0,0 +1,85 @@
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setNormalizer(
```
I still think there should be a better way instead of using normalizer. :stuck_out_tongue:

sarjon commented on src/PrestaShopBundle/Form/Validator/Constraints/UniqueStateIsoCodeValidator.php:
```diff
@@ -0,0 +1,91 @@
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Validator\Constraints;
```
Maybe it should be moved to `PrestaShop\PrestaShop\Core\ConstraintValidator` instead, not sure. :/

sarjon commented on src/PrestaShopBundle/Form/Validator/Constraints/UniqueStateIsoCodeValidator.php:
```diff
@@ -0,0 +1,91 @@
+/**
+ * Unique state iso code validator
+ */
+class UniqueStateIsoCodeValidator extends ConstraintValidator
```
I guess it could be `final`.

sarjon commented on src/PrestaShopBundle/Resources/config/routing/admin/improve/international/state.yml:
```diff
@@ -1,6 +1,35 @@
 admin_country_states:
   path: /
   methods: [GET]
+  condition: "request.isXmlHttpRequest()"
```
I know the reason behind this, but others may not, I suggest adding a little explanation above.

sarjon commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Locations/State/Blocks/form.html.twig:
```diff
@@ -0,0 +1,112 @@
+      {{ ps.form_group_row(stateForm.active, {}, {
+        'label': 'Status'|trans({}, 'Admin.Global'),
+      }) }}
+    </div>
```
Don't forget to render `form_rest`.

sarjon commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Locations/State/add.html.twig:
```diff
@@ -0,0 +1,38 @@
+
+{% block javascripts %}
+  {{ parent() }}
+{% endblock %}
```
You don't need this block.

sarjon commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Locations/State/edit.html.twig:
```diff
@@ -0,0 +1,40 @@
+
+{% block javascripts %}
+  {{ parent() }}
+{% endblock %}
```
Same here.

RaimondasSapola committed:
CR fixes

RaimondasSapola committed:
CR fixes

RaimondasSapola committed:
CR fixes

RaimondasSapola committed:
Added comment for normalizer usage

RaimondasSapola committed:
code style fixes

RaimondasSapola committed:
Minor cs changes

RaimondasSapola committed:
CR fixes

RaimondasSapola committed:
CS fixes

RaimondasSapola committed:
CR fixes

BlackKerio committed:
After modifying a product by webservice, the product position in the categories changes, even if we will use current position.

RaimondasSapola committed:
Minor validation update

BlackKerio committed:
Update classes/Product.php

Co-Authored-By: Pablo Borowicz <pablo.borowicz@prestashop.com>

BlackKerio committed:
Avoids webservice to create duplicated customers with same email

Commit:
fix: prevent order controller address validation from validating all guest addresses during guest checkout, should only apply to customers

Commit:
fix: coding style update

Commit:
Remove new-theme assets

Commit:
Remove admin default assets

Commit:
Remove themes assets

Commit:
Remove classic theme assets

Commit:
Create make file to install prestashop and build assets with 'make install' command

Commit:
Build assets in travis

BlackKerio committed:
Product.php

After the product has been modified by a webservice, while we also use associations, position in the categories changes

BlackKerio committed:
Update classes/Customer.php

Co-Authored-By: Pablo Borowicz <pablo.borowicz@prestashop.com>

BlackKerio committed:
Update Customer.php

LouiseBonnard committed:
Add Bangladesh localization pack

LouiseBonnard committed:
Add Bosnia and Herzegovina localization pack

LouiseBonnard committed:
Add Cameroon localization pack

LouiseBonnard committed:
Add Senegal localization pack

LouiseBonnard committed:
Add Madagascar localization pack

Commit:
Fixing submenu not showing on classic theme

Commit:
Adjusting quantity on product page if empty or under min qty

tdavidsonas88 committed:
Behat AddRootCategoryHandler

tdavidsonas88 committed:
Behat DeleteCategoryCoverImageHandler

tdavidsonas88 committed:
Behat fixes for EditRootCategory test

Commit:
Remove duplicate var

tdavidsonas88 committed:
Behat DeleteCategoryCoverImage

tdavidsonas88 committed:
cover image variable no longer needed

tdavidsonas88 committed:
Behat DeleteCategoryMenuThumbnailImage

tdavidsonas88 committed:
Behat DeleteCategoryMenuThumbnailImage missing test case

tdavidsonas88 committed:
Behat SetCategoryIsEnabled

tdavidsonas88 committed:
Behat BulkUpdateCategoriesStatusCommand

tdavidsonas88 committed:
Behat GetCategoryIsEnabledHandler

tdavidsonas88 committed:
fixes in code style

tdavidsonas88 committed:
commented out file uploads

tdavidsonas88 committed:
refactored the code after Tomas code review

tdavidsonas88 committed:
code style

jolelievre committed:
Display symbols in form in case of error, remove DefaultLanguage constraint

Commit:
Kick element from callback as he's not used

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

tdavidsonas88 committed:
ps_tmp_img_dir inject through services

tdavidsonas88 committed:
fixed category injection bugg

tdavidsonas88 committed:
fixes for image creation

tdavidsonas88 committed:
possible better ways to solve upload issues - not sure if injection of services has any value now / in the future

tdavidsonas88 committed:
full beauty of the parameters names

tdavidsonas88 committed:
getting temp image dir from the container in test

zuk3975 committed:
Fix suppliers form description to be optional

tdavidsonas88 committed:
refactored paths of images - chosen simpler solution

nesrineabdmouleh committed:
Add numbers for all directories

nesrineabdmouleh committed:
Fix selector id

SimonGrn committed:
Merge pull request #17201 from nesrineabdmouleh/fixEmailThemeSubTabSelector

Functional Tests - Fix "emailThemes/01_previewEmailThemes" test

SimonGrn committed:
Merge pull request #17202 from nesrineabdmouleh/functionaAddNumbersForAllTests

Functional tests - Add numbers for all directories

nesrineabdmouleh committed:
CRUD address

Commit:
Update contributors list for 1.7.6.3

nesrineabdmouleh committed:
Fis selector name

nesrineabdmouleh committed:
Fix identation

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-6

matks committed:
Merge pull request #17221 from matks/merge2develop-6

Merge 1.7.7.x into develop - 17/01/2020

nesrineabdmouleh committed:
enable disable friendly url

matks committed:
Recompute FOSJSRouting JSON file

jolelievre committed:
Merge pull request #17229 from matks/fix-177x

Recompute FOSJSRouting JSON file

matks committed:
Rebuild Create an Order page assets

matks committed:
Merge pull request #17232 from matks/fix-177x

 Rebuild Create an Order page assets

tdavidsonas88 committed:
PHP_UNIT_FRAMEWORK imported - was missing import after rebase

tdavidsonas88 committed:
using Assert instead of PHPUnit_Framework_Assert.

Commit:
Update configuration

Commit:
PHP cs fixer

Progi1984 committed:
Added RTL Style for Classic Theme

Commit:
Added RTL Style for Classic Theme (#17248)

Added RTL Style for Classic Theme

Progi1984 committed:
Changelog 1.7.6.3

Commit:
Changelog 1.7.6.3 (#17251)

Changelog 1.7.6.3

Commit:
Add Php-cs-fixer into github action

atomiix committed:
Fix deleted products not being displayed in orders

matks committed:
Merge pull request #17246 from PierreRambaud/cs-fixer-update

Update Cs Fixer configuration

matks committed:
Clean a bit files in install folder

matks committed:
Fix some links with http and no https

matthieu-rolland committed:
Merge pull request #17255 from matks/clean-install

Clean a bit files in install folder

tdavidsonas88 committed:
rolled back to $this->configuration->get('_PS_CAT_IMG_DIR_') in handler

matks committed:
Merge pull request #17267 from PrestaShop/1.7.6.x

Merge 1.7.6.x into 1.7.7.x - 21/01/2020

Progi1984 committed:
Merge pull request #17180 from zuk3975/m/supplier/fix-description

Fix suppliers form description to be optional

Progi1984 committed:
Merge pull request #16965 from NeOMakinG/issue10125

Adjusting quantity on product page if empty or under min qty

Progi1984 committed:
Merge pull request #16927 from LouiseBonnard/develop

Add BA, BD, CM, MG and SN localization packs

nesrineabdmouleh committed:
Limit the legal free text length to 50

PierreRambaud committed:
Merge pull request #17257 from matks/fix-https

Fix some links with http and no https

SimonGrn committed:
Merge pull request #17272 from nesrineabdmouleh/fixCheckFooterTextInPDF

Functional tests - Limit the legal free text length to 50

nesrineabdmouleh committed:
Add some fixes

atomiix committed:
Explain data population from OrderDetail

SimonGrn committed:
Merge pull request #17216 from nesrineabdmouleh/functionalCrudAddress

Functional tests - CRUD customer address

nesrineabdmouleh committed:
Address bulk actions

nesrineabdmouleh committed:
Use the same variable addressData

matks committed:
Add new badges to README

nesrineabdmouleh committed:
Fix bulkActionsDeleteButton selector

matks committed:
Enable bulk delete modal for Suppliers, Brands, Brand adresses

matks committed:
Apply php-cs-fixer

SimonGrn committed:
Merge pull request #17281 from nesrineabdmouleh/functionalAddressBulkActions

Functional tests - Addresses bulk actions

Progi1984 committed:
Merge pull request #17043 from jolelievre/fix-new-alternative-currency

Display symbols in form in case of error, remove DefaultLanguage constraint

matks committed:
Merge pull request #17284 from matks/add-badges

Add new badges to README

jolelievre committed:
Display symbols in form in case of error, remove DefaultLanguage constraint

PierreRambaud committed:
Merge pull request #16934 from NeOMakinG/issue16932

Fixing submenu not showing on classic theme

jolelievre committed:
Merge pull request #16670 from PierreRambaud/remove-useless-assets

Remove useless assets

atomiix committed:
Fix deleted products not being displayed in orders (#17254)

Fix deleted products not being displayed in orders

PierreRambaud committed:
Merge pull request #17291 from jolelievre/currency-symbols-error

Display symbols in form in case of error, remove DefaultLanguage constraint

matks committed:
Merge pull request #17286 from matks/enable-bulk-delete-modal-1

Enable bulk delete modal for Suppliers, Brands, Brand adresses

matks committed:
Adapt test for negative amount for money

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-8

matks committed:
Apply php-cs-fixer

matks committed:
Remove asset, they are not committed anymore on develop

jolelievre committed:
Merge pull request #17304 from matks/merge2develop-8

Merge 1.7.7.x into develop - 22/01/2020

nesrineabdmouleh committed:
Use lowercase

SimonGrn committed:
Merge pull request #17301 from nesrineabdmouleh/functionalEnableDisableFriendlyURL

Functional tests - Enable/Disable friendly url

tdavidsonas88 committed:
code style fix

Commit:
Build assets for nightly

Commit:
Remove github action to use PrettyCI only

nesrineabdmouleh committed:
Fix delete with modal in brands test

nesrineabdmouleh committed:
Fix delete with modal in suppliers test

nesrineabdmouleh committed:
Fix validation message in seo page test

nesrineabdmouleh committed:
Fix validation message

SimonGrn committed:
Merge pull request #17319 from PierreRambaud/fix/nightly-assets

Build assets for nightly

matks committed:
Merge pull request #17018 from tdavidsonas88/feature/category-part-2=integration-tests

Cover Category CommandHandlers and QueryHandlers by behat tests  - Part 2

matks committed:
Merge pull request #17328 from PierreRambaud/fix/prettyci

Remove PHP GitHub Actions and use PrettyCI only

Commit:
Remove share buttons after install

Commit:
Add make assets while creating release

SimonGrn committed:
Merge pull request #17329 from nesrineabdmouleh/fixClickBulkDeleteModal

Functional tests - Fix delete with bulk actions (brand, brand address, supplier) tests

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2-develop-9

BlackKerio committed:
Merge branch 'develop' into patch-4

BlackKerio committed:
Update Product.php

BlackKerio committed:
Merge branch 'develop' into fix-webservice-duplicate-email

Progi1984 committed:
Merge pull request #16697 from BlackKerio/patch-4

Fix setWsPositionInCategory and setWsCategories, the product position in the categories changes

Commit:
Prevent order controller from validating all guest addresses (#16620)

Prevent order controller from validating all guest addresses

Progi1984 committed:
Merge pull request #17330 from PierreRambaud/fix/update-build-process

Add make assets while creating release

tdavidsonas88 committed:
style fixes and correction of missing state_iso_code after rebase

matks committed:
Merge pull request #17334 from matks/merge2-develop-9

Merge 1.7.7.x into develop - 23/01/2020

tdavidsonas88 committed:
created and used Country VO for choice provider

tdavidsonas88 committed:
refactored country array to collection of Country VO

tdavidsonas88 committed:
AddStateHandlerInterface.php does not add value - removed it

tdavidsonas88 committed:
Query buddy refactored to queryBus

tdavidsonas88 committed:
only redirect for listing page if $e is an instance of StateNotFoundException

PierreRambaud committed:
Merge pull request #17340 from eternoendless/remove-installer-share-buttons

Remove share buttons after install

PierreRambaud committed:
Update PULL_REQUEST_TEMPLATE.md

matks committed:
Merge pull request #17369 from PrestaShop/update-github-template

Update PULL_REQUEST_TEMPLATE.md

matks committed:
Merge pull request #16553 from BlackKerio/fix-webservice-duplicate-email

Avoids webservice to create duplicated customers with same email

tdavidsonas88 committed:
Behat for AddStateHandler, GetStateForEditingHandler

tdavidsonas88 committed:
refactored state validation to throw more specific message

tdavidsonas88 committed:
reverted back AddStateHandlerInterface after Raimondas review - will be needed in the future

tdavidsonas88 committed:
simplified ConfigurableCountryChoiceType configureOptions not to use setNormalizer

tdavidsonas88 committed:
'id_country', ConfigurableCountryChoiceType placeholder refactored to use stand symfony form type option

tdavidsonas88 committed:
styles fixes

tdavidsonas88 committed:
simplified zone choice type rendering

tdavidsonas88 committed:
to avoid same urls changed '/' to '/list'

tdavidsonas88 committed:
comment changed for GetCountriesHandler.php

tdavidsonas88 committed:
simplified method after Julius recommendation

tdavidsonas88 committed:
after Julius code review recommendations

tdavidsonas88 committed:
after Julius code review implemented different link for ajax

tdavidsonas88 committed:
after Julius code review - follow namespace for service declaration

tdavidsonas88 committed:
annotation for PS standard

tdavidsonas88 committed:
Julius code review changes

tdavidsonas88 committed:
fix-cs

tdavidsonas88 committed:
missing type for new class

tdavidsonas88 committed:
instead of GetStateWithIsoCodeExistsHandler created CountryStateByIsoCodeProvider

tdavidsonas88 committed:
changed to prestashop_ui_kit and form_widget rendering
